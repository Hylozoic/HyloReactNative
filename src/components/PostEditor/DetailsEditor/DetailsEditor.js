import React from 'react'
import { Button, View } from 'react-native'
import KeyboardFriendlyView from '../../KeyboardFriendlyView'
import Editor from '../../Editor'
import { keyboardAvoidingViewProps as kavProps } from 'util/viewHelpers'
import { caribbeanGreen } from 'style/colors'
import styles from './DetailsEditor.styles'
import BackArrow from '../../BackArrow'

export default class DetailsEditor extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      headerTitle: 'Details',
      headerTitleStyle: { color: 'black' },
      headerTintColor: caribbeanGreen,
      headerLeft: <BackArrow navigation={navigation} />,
      headerRight: <View style={styles.saveButton}>
        <Button color={caribbeanGreen} title={'Save'} onPress={() => navigation.goBack()} />
      </View>
    }
  }

  constructor (props) {
    super(props)
    this.state = {
      content: props.initialContent
    }
  }

  componentWillUnmount () {
    this.props.saveChanges(this.state.content)
  }

  render () {
    const { initialContent, navigation } = this.props
    return <KeyboardFriendlyView style={styles.container} {...kavProps}>
      <Editor ref={ref => { this.editor = ref }}
        initialContent={initialContent}
        navigation={navigation}
        onChange={content => this.setState({content})}
        communityId={navigation.state.params.communityId} />
    </KeyboardFriendlyView>
  }
}
