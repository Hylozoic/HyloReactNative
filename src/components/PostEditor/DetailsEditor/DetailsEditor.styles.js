export default {
  saveButton: {
    marginRight: 10
  },
  container: {
    backgroundColor: 'white',
    flex: 1
  },
  backIcon: {
    height: 25,
    width: 15,
    marginLeft: 10
  },
  headerTitleStyle: {
    color: 'black'
  }
}
