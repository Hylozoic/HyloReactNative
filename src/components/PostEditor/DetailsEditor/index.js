import component from './DetailsEditor'
import connector from './DetailsEditor.connector'
export default connector(component)
