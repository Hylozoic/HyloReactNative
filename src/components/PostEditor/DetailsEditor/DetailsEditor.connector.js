import { connect } from 'react-redux'
import { setDetails } from '../PostEditor.store'

function mapStateToProps (state, props) {
  return {
    content: state.PostEditor.details
  }
}

function mapDispatchToProps (dispatch, props) {
  return {
    saveChanges: details => dispatch(setDetails(details))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
