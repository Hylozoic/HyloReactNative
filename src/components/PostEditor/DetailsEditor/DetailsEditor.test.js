import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import DetailsEditor from './DetailsEditor'

jest.mock('react-native-device-info')

const navigation = {
  state: {
    params: {
      communityId: 1
    }
  }
}
describe('DetailsEditor', () => {
  it('renders correctly', () => {
    const renderer = new ReactShallowRenderer()
    const initialContent = 'initial content'
    renderer.render(<DetailsEditor
      initialContent={initialContent}
      saveChanges={jest.fn()}
      navigation={navigation}
    />)
    const actual = renderer.getRenderOutput()
    expect(actual).toMatchSnapshot()
  })

  it('has navigation options', () => {
    const props = {navigation: {state: {}}}
    expect(DetailsEditor.navigationOptions(props)).toMatchSnapshot()
  })
})
