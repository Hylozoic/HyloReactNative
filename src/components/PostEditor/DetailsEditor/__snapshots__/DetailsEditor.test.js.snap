// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsEditor renders correctly 1`] = `
<Unknown
  behavior="padding"
  keyboardVerticalOffset={64}
>
  <Editor
    communityId={1}
    initialContent="initial content"
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "communityId": 1,
          },
        },
      }
    }
    onChange={[Function]}
  />
</Unknown>
`;
